internal/envoy/v3/runtime.go

```diff
@@ -36,8 +36,11 @@ func RuntimeLayers() []*envoy_service_runtime_v3.Runtime {
 func baseRuntimeLayer() *structpb.Struct {
 	return &structpb.Struct{
 		Fields: map[string]*structpb.Value{
-			"re2.max_program_size.error_level": {Kind: &structpb.Value_NumberValue{NumberValue: maxRegexProgramSizeError}},
-			"re2.max_program_size.warn_level":  {Kind: &structpb.Value_NumberValue{NumberValue: maxRegexProgramSizeWarn}},
+			"re2.max_program_size.error_level":                     structpb.NewNumberValue(maxRegexProgramSizeError),
+			"re2.max_program_size.warn_level":                      structpb.NewNumberValue(maxRegexProgramSizeWarn),
+			"overload.premature_reset_total_stream_count":          structpb.NewNumberValue(50),
```

Comment:
This and the below have defaults that Envoy sets so I'm not sure we need to set these ourselves. 50 also is a lot lower than the Envoy default of 500, can you offer some reasoning behind setting this to 50?

see https://www.envoyproxy.io/docs/envoy/v1.27.1/version_history/v1.27/v1.27.1.html

Reply:
I tested perf during the attack by setting `premature_reset_total_stream_count` to 50,100 and 500.
Roughly, Envoy served 4 times more RPS for legitimate traffic when value was set to `50` vs `500`.

Reply:
In that case, I think we can follow up on https://github.com/projectcontour/contour/pull/5827 with a change to make this runtime field configurable, as it seems it is something that needs tuning to a particular environment/usage

PRs welcome for that change

Reply:
added https://github.com/projectcontour/contour/issues/5847 to cover this